### PR TITLE
drainer: filter out placement special comments in ddl

### LINKED ISSUE #1145: Filter out placement special comments when executing ddl in drainer
## Feature Request

When syncing the a tidb downstream, we are using `tidb_placement_mode='ignore'` to skip all placement settings. However, for some tidb version like `5.0` , `tidb_placement_mode` not exists but placement also works. Some error may occurs when the downsteam is `5.0`.

So we should filter placement special comments in sql to avoid this.

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #1145

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### Release note

- No release note

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

**Comment by lance6716** [comment]
can we use parser to remove the comment?

File: pkg/loader/load.go [hunk 1]
```diff
@@ -418,9 +418,18 @@ func (s *loaderImpl) processMysqlDDL(ddl *DDL) error {
...
 
-		if _, err = tx.Exec(ddl.SQL); err != nil {
+		sql, err := removeDDLPlacementOptions(ddl.SQL)
+		if err != nil {
+			return err
+		}
+
+		if sql == "" {
+			log.Info("skip ddl with all placement options", zap.String("sql", ddl.SQL))
+		}
```
Comment (lichunzhu) [inline pkg/loader/load.go:428]: return `tx.Commit` here?

**Comment by lichunzhu** [comment]
> can we use parser to remove the comment?

+1

**Comment by lcwangchao** [comment]
> can we use parser to remove the comment?

I tried to use parser, but it should first the upgrade mode. However upgrading the depencies casuses more problems and some other ut fails...

File: pkg/loader/util.go [hunk 2]
```diff
@@ -448,3 +449,18 @@ func getOracleUniqKeys(db *gosql.DB, schema, table string) (uniqueKeys []indexIn
...
 }
+
+func removeDDLPlacementOptions(sql string) (string, error) {
+	stmt, err := parser.New().ParseOneStmt(sql, "", "")
+	if err != nil {
+		return "", err
+	}
+
+	var sb strings.Builder
+	restoreCtx := format.NewRestoreCtx(format.DefaultRestoreFlags|format.RestoreTiDBSpecialComment|format.SkipPlacementRuleForRestore, &sb)
```
Comment (lichunzhu) [inline pkg/loader/util.go:460]: Can we check whether to change DDL by parser checking does this DDL contains `placementRule`?
Maybe most DDLs don't contain placement rules. But after this commit, their DDLs will be all rewritten.

Comment (lcwangchao) [reply pkg/loader/util.go:460]: The simplest way is to check if string contains the string `/*T![placement]`

File: pkg/loader/load.go [hunk 3]
```diff
@@ -418,9 +418,19 @@ func (s *loaderImpl) processMysqlDDL(ddl *DDL) error {
 			}
 		}
 
-		if _, err = tx.Exec(ddl.SQL); err != nil {
+		sql, err := removeDDLPlacementOptions(ddl.SQL)
+		if err != nil {
+			return err
+		}
```
Comment (lichunzhu) [inline pkg/loader/load.go:424]: Can we just warn this error here and execute the origin SQL? just like we did in
https://github.com/pingcap/tidb-binlog/blob/194d4ac/pkg/util/util.go#L343

**Review by lichunzhu — APPROVED** [review]
Rest LGTM

File: pkg/loader/load.go (hunk 3 above)
Comment (lcwangchao) [reply pkg/loader/load.go:424]: fine

**Review by lance6716 — COMMENTED** [review]
lgtm, I'll approve after https://github.com/pingcap/tidb-binlog/pull/1144#discussion_r829782348 is resolved

File: pkg/loader/load.go (hunk 3 above)
Comment (lcwangchao) [reply pkg/loader/load.go]: done

**Review by lance6716 — APPROVED** [review]

**Pull request merged**